app/controllers/concerns/oai_concern.rb

```diff
@@ -51,7 +51,35 @@ def message
     end
   end
 
+  # Token for requesting records: base64-encoded combo of filters & page cursor
+  # A token lets you construct a list of records and point to somewhere in that
+  # list.
+  class ResumptionToken
+    def self.encode(set: nil, page: nil, from_date: nil, until_date: nil)
+      validate(set, page, from_date, until_date)
+      Base64.urlsafe_encode64([set, page, from_date, until_date].join(';'))
```

Comment:
I'd suggest we add some kind of version string (e.g. `v1`)  at the start of this, in case we need to change the format in the future.

Reply:
Great suggestion; added